src/consensus/params.cpp

```diff
@@ -0,0 +1,56 @@
+#include "params.h"
+
+#include "main.h"
+#include "upgrades.h"
+
+namespace Consensus {
+    bool Params::NetworkUpgradeActive(int nHeight, Consensus::UpgradeIndex idx) const {
+        return NetworkUpgradeState(nHeight, *this, idx) == UPGRADE_ACTIVE;
+    }
+
+    int Params::GetLastFoundersRewardBlockHeight(int nHeight) const {
+        // zip208
+        // FoundersRewardLastBlockHeight := max({ height ⦂ N | Halving(height) < 1 })
+        // H := blossom activation height; SS := SubsidySlowStartShift(); R := BLOSSOM_POW_TARGET_SPACING_RATIO
+        // preBlossom:
+        // 1 = (height - SS) / preInterval
+        // height = preInterval + SS
+        // postBlossom:
+        // 1 = (H - SS) / preInterval + (height - H) / postInterval
+        // height = H + postInterval + (SS - H) * (postInterval / preInterval)
+        // height = H + postInterval + (SS - H) / R
```

Comment:
There is an error in this derivation and the code that follows. R := preSpacing / postSpacing = postInterval / preInterval.
This should be `... * R` not `... / R`

Reply:
Resolved this in https://github.com/zcash/zcash/pull/4025/commits/1188c9adf7bcbecabe9bd70357d76992872f486b